modules/programs/firefox/mkFirefoxModule.nix

```diff
@@ -343,410 +343,421 @@ in {
 
     profiles = mkOption {
       inherit visible;
-      type = types.attrsOf (types.submodule ({ config, name, ... }: {
-        options = {
-          name = mkOption {
-            type = types.str;
-            default = name;
-            description = "Profile name.";
-          };
+      type = types.attrsOf (types.submodule ({ config, name, ... }:
+        let
 
-          id = mkOption {
-            type = types.ints.unsigned;
-            default = 0;
-            description = ''
-              Profile ID. This should be set to a unique number per profile.
-            '';
-          };
-
-          settings = mkOption {
-            type = types.attrsOf (jsonFormat.type // {
-              description =
-                "${name} preference (int, bool, string, and also attrs, list, float as a JSON string)";
-            });
-            default = { };
-            example = literalExpression ''
-              {
-                "browser.startup.homepage" = "https://nixos.org";
-                "browser.search.region" = "GB";
-                "browser.search.isUS" = false;
-                "distribution.searchplugins.defaultLocale" = "en-GB";
-                "general.useragent.locale" = "en-GB";
-                "browser.bookmarks.showMobileBookmarks" = true;
-                "browser.newtabpage.pinned" = [{
-                  title = "NixOS";
-                  url = "https://nixos.org";
-                }];
-              }
-            '';
-            description = ''
-              Attribute set of ${name} preferences.
+          profile = config;
```

Comment:
nit: It would cleanup the diff a lot if you just referred to `config` directly rather than giving it an alias of profile, I could see why you'd want to do this to avoid confusion with the top-level `config` though.

Reply:
Agreed, a decent middle way may be to introduce the `profile = config` binding inside the `config` attribute. I.e.,

```nix
config = let profile = config; in {
  settings."toolkit.legacyUserProfileCustomizations.stylesheets" =
    mkIf (profile.userChrome != "" || profile.userContent != "") true;
};
```

Reply:
You're maintainers, so what do you think it's better for maintainability? Using it only in `config` means, that if in the future it won't be possible to refer to `profile` in `settings` in case it's needed e.g. for some `default`.

The bad things are that it will mess up any other PR's that are open right now, and it will also mess up `git blame` (though, there is `git blame -w`, which ignores those indentation changes, not idea whether github uses it).

Reply:
I'd say move the let to the config attribute as in my previous comment. We can deal with the (probably unlikely) event that the profile variable needs to be referenced in the options attribute when it happens 🙂

Reply:
Done